analyzers/src/SonarAnalyzer.CSharp/Rules/BeginInvokePairedWithEndInvoke.cs

```diff
@@ -60,14 +60,30 @@ public sealed class BeginInvokePairedWithEndInvoke : BeginInvokePairedWithEndInv
         protected override void VisitInvocation(EndInvokeContext context) =>
             new InvocationWalker(context).SafeVisit(context.Root);
 
+        protected override bool IsInvalidCallback(SyntaxNode callbackArg, SemanticModel semanticModel)
+        {
+            if (FindCallback(callbackArg, semanticModel) is { } callback)
+            {
+                if (Language.Syntax.Kind(callback) == SyntaxKind.SimpleMemberAccessExpression)
+                {
+                    return !(semanticModel.GetSymbolInfo(callback).Symbol is IMethodSymbol methodSymbol
```

Comment:
I think we can first check the name at the syntax level and only later get the symbol, in order to improve performance.